src/consumer.cpp

```diff
@@ -35,7 +37,36 @@ IceflowConsumer::IceflowConsumer(std::shared_ptr<IceFlow> iceflow,
 IceflowConsumer::~IceflowConsumer() { unsubscribeFromAllPartitions(); }
 
 std::vector<uint8_t> IceflowConsumer::receiveData() {
-  return m_inputQueue.waitAndPopValue();
+  auto value = m_inputQueue.waitAndPopValue();
+
+  // TODO: Discuss if this should be put elsewhere
+  saveTimestamp();
```

Comment:
Should be fine to keep it here.

```suggestion
  saveTimestamp();
```